### PR TITLE
feat(codepipeline): allow creation of pipelines without source trigger

### LINKED ISSUE #1652: codepipeline: GitHubSourceAction should allow setup without polling or webhook
codepipeline.GitHubSourceAction takes a 'pollForSourceChanges' setting. If it is true, codepipeline will poll github for changes. If it is false, then cdk will automatically create a webhook and configure the pipeline use it as a trigger.

This is almost always the right thing to do.

But if you want to trigger a pipeline on a schedule, it would be great to be able to create a pipeline with both of these triggers disable.

There is a workaround to achieve this:

```
new codepipeline.GitHubSourceAction(this, construct_base_name + 'GitHubSource', {
    ... other settings ...
    // this is needed to prevent webhook creation, but we'll disable it later
    pollForSourceChanges: true
})

const cfnPipeline = pipeline.node.findChild('Resource') as codepipeline.CfnPipeline  		      cfnPipeline.addPropertyOverride('Stages.0.Actions.0.Configuration.PollForSourceChanges', false)
```

### PR DESCRIPTION
**Pull request by SanderKnape**
Fixes #1652 

----

### Pull Request Checklist

* [ ] Testing
  - Unit test added (prefer not to modify an existing test, otherwise, it's probably a breaking change)
  - __CLI change?:__ coordinate update of integration tests with team
  - __cdk-init template change?:__ coordinated update of integration tests with team
* [ ] Docs
  - __jsdocs__: All public APIs documented
  - __README__: README and/or documentation topic updated
* [ ] Title and Description
  - __Change type__: title prefixed with **fix**, **feat** will appear in changelog
  - __Title__: use lower-case and doesn't end with a period
  - __Breaking?__: last paragraph: "BREAKING CHANGE: <describe what changed + link for details>"
  - __Issues__: Indicate issues fixed via: "**Fixes #xxx**" or "**Closes #xxx**"
* [ ] Sensitive Modules (requires 2 PR approvers)
  - IAM Policy Document (in @aws-cdk/aws-iam)
  - EC2 Security Groups and ACLs (in @aws-cdk/aws-ec2)
  - Grant APIs (only if not based on official documentation with a reference)

----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

----

Notes

* My first PR here, I went through the CONTRIBUTING file but I certainly may have missed something
* I changed the existing `github action uses ThirdParty owner` test to verify that the default behaviour has remained the same
* Since there are now three options possible with the replaced property, I added two more test to include all use cases

----

BREAKING CHANGE: this changes the GitHub source action `pollForSourceChanges` property to the `trigger` property that allows `Webhook` (default), `Poll` and `None` (new)

### CODE REVIEW

File: packages/@aws-cdk/aws-codepipeline-actions/test/test.pipeline.ts [hunk 1]
```diff
@@ -65,6 +65,204 @@ export = {
...
+          "InputArtifacts": [],
+          "Name": "Boo",
+          "OutputArtifacts": [],
+          "RunOrder": 1
+        }
+        ],
+        "Name": "Two"
+      }
+      ]
+    }));
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codepipeline-actions/test/test.pipeline.ts]: When you're using `haveResourceLike`, you don't need all these extra properties. Focus on just the ones important for your test, something like:

```ts
    expect(stack).to(haveResourceLike('AWS::CodePipeline::Pipeline', {
      "Stages": [
        {
          "Actions": [
            {
              "Name": "GH",
              "Configuration": {
                "PollForSourceChanges": true
              },
            }
          ],
          "Name": "Source"
        },
        {
          "Actions": [
            {
              "Name": "Boo"
            }
          ],
          "Name": "Two"
        }
      ]
    }));
```

File: packages/@aws-cdk/aws-codepipeline-actions/test/test.pipeline.ts [hunk 2]
```diff
@@ -65,6 +65,204 @@ export = {
...
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codepipeline-actions/test/test.pipeline.ts]: Same comment here as above.

File: packages/@aws-cdk/cfnspec/tsconfig.tsbuildinfo [hunk 3]
```diff
@@ -1,431 +1,431 @@
 {
   "program": {
     "fileInfos": {
-      "/users/benisrae/code/cdk/aws-cdk-4/tools/cdk-build-tools/node_modules/typescript/lib/lib.es5.d.ts": {
```
Comment (skinny85) [inline packages/@aws-cdk/cfnspec/tsconfig.tsbuildinfo]: This was committed accidentally, and should not be included in the PR.

**Comment by SanderKnape** [comment]
Thanks for the feedback @skinny85. I have implemented your requested changes - let me know if there is anything else I need to change!
